### PR TITLE
Feature/delete data logic

### PR DESCRIPTION
**Pull request by ArtemisSoftware**

### CODE REVIEW

File: app/src/main/java/com/artemissoftware/tasky/agenda/presentation/dashboard/AgendaViewModel.kt [hunk 1]
```diff
@@ -193,9 +197,10 @@ class AgendaViewModel @Inject constructor(
         }
     }
 
+    @OptIn(FlowPreview::class)
```
Comment (MKiperszmid) [inline app/src/main/java/com/artemissoftware/tasky/agenda/presentation/dashboard/AgendaViewModel.kt:200]: Why do you need this annotation tho?

Comment (ArtemisSoftware) [reply app/src/main/java/com/artemissoftware/tasky/agenda/presentation/dashboard/AgendaViewModel.kt:200]: It was suggested by the IDE

File: app/src/main/java/com/artemissoftware/tasky/agenda/domain/usecase/agenda/SyncAgendaPeriodicallyUseCase.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import com.artemissoftware.tasky.agenda.domain.sync.AgendaSynchronizer
+import java.time.LocalDate
+import java.util.UUID
+import javax.inject.Inject
+
+class SyncAgendaPeriodicallyUseCase @Inject constructor(
+    private val agendaSynchronizer: AgendaSynchronizer,
+) {
+    operator fun invoke(date: LocalDate): UUID {
+        return agendaSynchronizer.syncAgenda(currentDate = date)
```
[the remark this answers is not in the document]
Comment (ArtemisSoftware) [reply app/src/main/java/com/artemissoftware/tasky/agenda/domain/usecase/agenda/SyncAgendaPeriodicallyUseCase.kt:12]: When I change the weekday I want to sync that particular day.

I was checking the documentation I did not find that endpoint.
According to the documentation there is only one method to get the agenda and It says

`Returns the agenda for a given day`

File: app/src/main/java/com/artemissoftware/tasky/agenda/data/remote/sync/AgendaSynchronizerImpl.kt [hunk 3]
```diff
@@ -1,24 +1,53 @@
...
+import java.time.LocalDate
 import java.util.UUID
 import java.util.concurrent.TimeUnit
 
 class AgendaSynchronizerImpl(
     private val workManager: WorkManager,
 ) : AgendaSynchronizer {
 
+    override fun syncAgenda(currentDate: LocalDate): UUID {
+        workManager.cancelAllWorkByTag("syncAgenda")
```
[the remark this answers is not in the document]
Comment (ArtemisSoftware) [reply app/src/main/java/com/artemissoftware/tasky/agenda/data/remote/sync/AgendaSynchronizerImpl.kt:20]: Actually ExistingPeriodicWorkPolicy.REPLACE is deprecated the suggestion is to use UPDATE.
I will update the Work and not restart it

File: app/src/main/java/com/artemissoftware/tasky/agenda/data/remote/sync/AgendaSynchronizerImpl.kt [hunk 4]
```diff
@@ -1,24 +1,53 @@
...
+            TimeUnit.MINUTES,
+        )
+            .setConstraints(
+                Constraints.Builder()
+                    .setRequiredNetworkType(NetworkType.CONNECTED)
+                    .build(),
+            )
+            .setInputData(
+                Data.Builder()
+                    .putLong(WorkerKeys.SELECTED_DATE, currentDate.toEpochMilli())
```
Comment (philipplackner) [inline app/src/main/java/com/artemissoftware/tasky/agenda/data/remote/sync/AgendaSynchronizerImpl.kt:35]: Why do you need the selected date here? In the periodic worker you're syncing the full agenda

File: app/src/main/java/com/artemissoftware/tasky/agenda/domain/usecase/agenda/SyncAgendaPeriodicallyUseCase.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (philipplackner) [reply app/src/main/java/com/artemissoftware/tasky/agenda/domain/usecase/agenda/SyncAgendaPeriodicallyUseCase.kt:12]: There is a route which gives you all agenda items for a user which is intended for syncing and it's contained in the API docs. It's called /fullAgenda.
I added this after the last round, but few weeks before this one started, maybe you joined a few days before I updated the PDF, so can you re-download it from elopage?

**Pull request merged**